db/db_impl/db_impl.h

```diff
@@ -1965,6 +1969,11 @@ class DBImpl : public DB {
   void ReleaseFileNumberFromPendingOutputs(
       std::unique_ptr<std::list<uint64_t>::iterator>& v);
 
+  // Similar to pending_outputs, preserve OPTIONS file for remote compaction
+  std::list<uint64_t>::iterator CaptureOptionsFileNumberForRemoteCompaction();
+  void ReleaseOptionsFileNumberFromRemoteCompaction(
```

Comment:
Nit: Can we just call these something like `CaptureCurrentOptionsFileNumber` instead of making it specific to remote compaction?